### PR TITLE
Fix acceptance tests by ignoring retryable errors

### LINKED ISSUE #4517: circleci: failed tests (13046): TestClusterRecovery
The following test appears to have failed:

[#13046](https://circleci.com/gh/cockroachdb/cockroach/13046):

```
I0219 16:55:12.282009 28384 acceptance/chaos_test.go:170  client 0 shutting down
I0219 16:55:12.287312 28384 acceptance/chaos_test.go:170  client 1 shutting down
I0219 16:55:12.287367 28384 acceptance/chaos_test.go:170  client 2 shutting down
W0219 16:55:12.337542 28384 acceptance/chaos_test.go:242  round 2: cluster recovered
I0219 16:55:12.437925 28384 acceptance/cluster/localcluster.go:583  stopping
--- FAIL: TestClusterRecovery (16.47s)
	chaos_test.go:147: dial tcp 0.0.0.0:32799: getsockopt: connection refused
=== RUN   TestNodeRestart
I0219 16:55:14.733030 28384 acceptance/cluster/localcluster.go:259  started docker-spy: 172.17.0.25
I0219 16:55:14.733056 28384 acceptance/cluster/localcluster.go:266  initializing cluster
I0219 16:55:16.353571 28384 acceptance/cluster/localcluster.go:519  creating certs (1024bit) in: /home/ubuntu/cockroach/.localcluster.certs.930358903
I0219 16:55:17.916638 28384 acceptance/cluster/localcluster.go:437  *** started roach0.local ***
  ui:    https://0.0.0.0:32801
  trace: https://0.0.0.0:32801/debug/requests
  logs:  /tmp/circle-artifacts.6eCohat/acceptance/TestNodeRestart/roach0.local/cockroach.INFO
  pprof: docker exec -it 0c91b /bin/bash -c 'go tool pprof /cockroach <(wget --no-check-certificate -qO- https://$(hostname):26257/debug/pprof/heap)'
--
--- SKIP: TestBuildBabyCluster (0.00s)
	terraform_test.go:32: only enabled during testing
=== RUN   TestFiveNodesAndWriters
--- SKIP: TestFiveNodesAndWriters (0.00s)
	main_test.go:63: running in docker mode
FAIL
ok github.com/cockroachdb/cockroach/acceptance 1337s

```
Please assign, take a look and update the issue accordingly.

### PR DESCRIPTION
**Pull request by vivekmenezes**
closes #4517

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5006)

### CODE REVIEW

File: acceptance/chaos_test.go [hunk 1]
```diff
@@ -134,18 +135,34 @@ func transferMoney(client *testClient, numAccounts, maxTransfer int) error {
...
+		InitialBackoff: 20 * time.Millisecond,
+		MaxBackoff:     100 * time.Millisecond,
+		Multiplier:     2,
 	}
-	if sum != 0 {
-		t.Fatalf("The bank is not in good order. Total value: %d", sum)
+	for loop := retry.Start(options); loop.Next(); {
+		// Hold the read lock on the client to prevent it being restarted by
+		// chaos monkey.
+		client.RLock()
```
Comment (tamird) [inline acceptance/chaos_test.go]: you're locking this many times in a loop; that seems incorrect

**Comment by vivekmenezes** [comment]
I changed this to use util.SucceedsSoon()

File: acceptance/chaos_test.go [hunk 2]
```diff
@@ -134,19 +135,31 @@ func transferMoney(client *testClient, numAccounts, maxTransfer int) error {
...
 	var sum int
-	client.RLock()
-	defer client.RUnlock()
-	if err := client.db.QueryRow("SELECT SUM(balance) FROM bank.accounts").Scan(&sum); err != nil {
-		t.Fatal(err)
-	}
+	util.SucceedsSoon(t, func() error {
+		// Hold the read lock on the client to prevent it being restarted by
+		// chaos monkey.
+		client.RLock()
```
Comment (BramGruneir) [inline acceptance/chaos_test.go]: Should this be outside the function?

**Comment by BramGruneir** [comment]
LGTM

**Comment by vivekmenezes** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[acceptance/chaos_test.go, line 152 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5006#-KCMsh8otIV3QorS5TYP-r1-152:-KCN06aDed_e_9lwuw0V:160423075) ([raw file](https://github.com/cockroachdb/cockroach/blob/1800d06b05663d08e0c2fcf415785bbc3c24b5d8/acceptance/chaos_test.go#L152)):*
why

---

*[acceptance/chaos_test.go, line 148 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5006#-KCMsh8otIV3QorS5TYP-r2-148:-KCN-vBH9XRihrmzAsEa:-1634245254) ([raw file](https://github.com/cockroachdb/cockroach/blob/0c2cec17e78087ad8332ebf3f0c1204c1caab1f7/acceptance/chaos_test.go#L148)):*
No this has to be inside inside because it protects client.db

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5006)*

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/chaos_test.go, line 154 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5006#-KCN50cwBQHncinZBzOq:-KCN50cxnvY2Z5rEBeYD:-332607399) ([raw file](https://github.com/cockroachdb/cockroach/blob/9fc6f5fefd8db1d350c9da494478e219cb412032/acceptance/chaos_test.go#L154)):*
remove this and always `return err` below

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5006#-:-KCN5AQCM3klW2XzUfbU:1417088721)*

**Comment by vivekmenezes** [comment]
I'm still seeing a few errors on CI. I'll ping the thread once I resolve them because we might need to skip a few more errors.

	chaos_test.go:271: read tcp 127.0.0.1:50009->127.0.0.1:32781: read: connection reset by peer
	chaos_test.go:271: EOF
	chaos_test.go:152: EOF

**Comment by vivekmenezes** [comment]
I had to add two more retryable errors to this PR:
EOF
connection reset by peer

Both are connection related errors. EOF is sent when a server shuts down cleanly. reset is when it comes back up and cannot recognize the client connection.

**Pull request merged**